StripeConnect/StripeConnect/Source/Internal/Webview/MessageHandlers/Helpers/ScriptMessageHandlerWithReply.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ScriptMessageHandlerWithReply.swift
+//  StripeConnect
+//
+//  Created by Mel Ludowise on 5/1/24.
+//
+
+import WebKit
+
+/// Convenience class that conforms to WKScriptMessageHandlerWithReply and can be instantiated with a closure
+class ScriptMessageHandlerWithReply<T: Decodable, R: Codable>: NSObject, WKScriptMessageHandlerWithReply {
+    let name: String
+    let didReceiveMessage: (T) async throws -> R
+    
+    init(name: String,
+         didReceiveMessage: @escaping (T) async throws -> R) {
+        self.name = name
+        self.didReceiveMessage = didReceiveMessage
+    }
+    
+    @MainActor
+    func userContentController(_ userContentController: WKUserContentController,
+                               didReceive message: WKScriptMessage) async -> (Any?, String?) {
+        guard message.name == name else {
+            debugPrint("Unexpected message name: \(message.name)")
+            return (nil, "Unexpected message")
+        }
+        guard
+            let bodyData = (message.body as? String)?.data(using: .utf8),
+            let body = try? JSONDecoder().decode(T.self, from: bodyData) else {
+            debugPrint("Failed to decode body for message with name: \(message.name)")
+            return (nil, "Failed to decode body for message")
+        }
+        do {
+            let value = try await didReceiveMessage(body)
+            let data = try JSONEncoder().encode(value)
+            guard let response = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) else {
+                return (nil, "Failed to decode body")
+            }
```

Comment:
optional – We could return the error message here instead of hardcoding the string. That said, I think the error message is just something that gets forwarded to JS, so it probably only matters if we're logging it from within the JS layer and want that information. We already plan to log errors from the mobile layer as part of [MXMOBILE-2491](https://jira.corp.stripe.com/browse/MXMOBILE-2491), so it probably doesn't matter too much.
```suggestion
        guard let bodyData = (message.body as? String)?.data(using: .utf8) else {
            return (nil, "Failed to decode body for message")
        }
        do {
            let body = try JSONDecoder().decode(T.self, from: bodyData)
            let value = try await didReceiveMessage(body)
            let data = try JSONEncoder().encode(value)
            let response = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
```